pytorch_lightning/core/hooks.py

```diff
@@ -684,22 +688,24 @@ def transfer_batch_to_device(self, batch, device):
             - :meth:`move_data_to_device`
             - :meth:`apply_to_collection`
         """
-        device = device or self.device
```

Comment:
Why remove this?
This is a useful function to have. Let's keep it optional?

Reply:
`transfer_batch_to_device` is public api. There the user should get the actual device, otherwise device is always.